src/Razor/benchmarks/Microsoft.AspNetCore.Razor.Microbenchmarks/LanguageServer/RazorDiagnosticsBenchmark.cs

```diff
@@ -58,79 +66,95 @@ public async Task SetupAsync()
+                        Source = "DocumentPullDiagnosticHandler",
+                        Message = "The name 'CallOnMe' does not exist in the current context"
+                    }).ToArray()
+                }
             }, Array.Empty<VSInternalDiagnosticReport>());
     }
 
     private protected override LanguageServerFeatureOptions BuildFeatureOptions()
     {
-        return new DiagnosticsLanguageServerFeatureOptions();
+        return Mock.Of<LanguageServerFeatureOptions>(options =>
+            options.SupportsFileManipulation == true &&
+            options.SupportsDelegatedCodeActions == true &&
+            options.SingleServerSupport == true &&
+            options.SingleServerCompletionSupport == true &&
+            options.CSharpVirtualDocumentSuffix == ".ide.g.cs" &&
+            options.HtmlVirtualDocumentSuffix == "__virtual.html",
+            MockBehavior.Strict);
     }
 
-    private string GetFileContents(FileTypes fileType)
+    private IRazorDocumentMappingService BuildRazorDocumentMappingService()
     {
-        var sb = new StringBuilder();
+        var razorDocumentMappingService = new Mock<IRazorDocumentMappingService>(MockBehavior.Strict);
+
+        Range? hostDocumentRange;
+        razorDocumentMappingService.Setup(
+            r => r.TryMapToHostDocumentRange(
+                It.IsAny<IRazorGeneratedDocument>(),
+                _inRange,
+                It.IsAny<MappingBehavior>(),
+                out hostDocumentRange))
+            .Returns((IRazorGeneratedDocument generatedDocument, Range range, MappingBehavior mappingBehavior, out Range? actualOutRange) =>
+            {
+                actualOutRange = _outRange;
+                return true;
+            });
+
+        Range? hostDocumentRange2;
+        razorDocumentMappingService.Setup(
+            r => r.TryMapToHostDocumentRange(
+                It.IsAny<IRazorGeneratedDocument>(),
+                It.IsNotIn(_inRange),
+                It.IsAny<MappingBehavior>(),
+                out hostDocumentRange2))
+            .Returns((IRazorGeneratedDocument generatedDocument, Range range, MappingBehavior mappingBehavior, out Range? actualOutRange) =>
+            {
+                actualOutRange = null;
+                return false;
+            });
 
-        sb.Append("""
-            @using System;
-            """);
+        return razorDocumentMappingService.Object;
+    }
 
-        for (var i = 0; i < (fileType == FileTypes.Small ? 1 : 100); i++)
-        {
-            sb.Append($$"""
-            @{
-                var y{{i}} = 456;
-            }
+    private ILoggerFactory BuildLoggerFactory() => Mock.Of<ILoggerFactory>(
+        r => r.CreateLogger(
+            It.IsAny<string>()) == new NoopLogger(),
+        MockBehavior.Strict);
 
-            <div>
-                <p>Hello there Mr {{i}}</p>
-            </div>
-            """);
-        }
+    private string GetFileContents(FileTypes fileType)
```

Comment:
Question: Is the file content even needed any more, if the document mapping service is mocked? The numbers in the benchmark seem to be pretty close for "small" versus "large"

Reply:
I'm looking into this

Reply:
Your observation is accurate.
Since I'm mocking the razor mapping service with the current setting the file size does not make a difference in the benchmarks.

I will conclude this PR by just having one benchmark that is mocking the razor mapping service.

Will have another PR that sets up razor mapping service and adds another benchmark with a larger file size and more diagnostics. That is gonna change the `RazorDiagnosticsBenchmark` quite a bit. So I'd better address that separately from this PR